File: typescript/functions/sendTransaction.ts
import {
  Network,
  SymbolAccount,
  SymbolFacade,
  descriptors,
} from "symbol-sdk/symbol"
import { awaitTransactionStatus } from "./awaitTransactionStatus" // 必要なインポートを追加

const NODE_URL = "https://sym-test-03.opening-line.jp:3001"
const facade = new SymbolFacade(Network.TESTNET)

export async function sendTransaction(
  descriptor: any,
  signAccount: SymbolAccount,
  description: string,
) {
  const tx = facade.createTransactionFromTypedDescriptor(
    descriptor,
    signAccount.publicKey, // 送信元アカウントの公開鍵
    100,
    60 * 60 * 2,
  )

  const signature = signAccount.signTransaction(tx) // 署名
  const jsonPayload =
    facade.transactionFactory.static.attachSignature(tx, signature) // ペイロード

  const response = await fetch(new URL("/transactions", NODE_URL), {
    method: "PUT",
    headers: { "Content-Type": "application/json" },
    body: jsonPayload,
  }).then((res) => res.json())

  console.log({ response })

  const hash = facade.hashTransaction(tx)

  console.log(`===${description}===`)
  await awaitTransactionStatus(hash.toString(), NODE_URL, "confirmed")
}

File: python/functions/send_transaction.py
import requests
from symbolchain.facade.SymbolFacade import (
  SymbolFacade,
  SymbolAccount,
  Hash256,
)
from typing import Any
from symbolchain.sc import Amount, Signature


# トランザクションを受け取り、署名し、トランザクションハッシュを返す関数
def send_transaction(tx: Any, signAccount: SymbolAccount) -> Hash256:
  NODE_URL: str = "https://sym-test-03.opening-line.jp:3001"
  facade: SymbolFacade = SymbolFacade("testnet")

  tx.fee = Amount(100 * tx.size)

  signature: Signature = signAccount.sign_transaction(tx)

  json_payload: str = facade.transaction_factory.attach_signature(
    tx, signature
  )

  response = requests.put(
    f"{NODE_URL}/transactions",
    headers={"Content-Type": "application/json"},
    data=json_payload,
  ).json()

  print("Response:", response)

  hash: Hash256 = facade.hash_transaction(tx)

  return hash
